plugins/scaffolder-backend/src/scaffolder/tasks/NunjucksWorkflowRunner.ts

```diff
@@ -355,6 +382,14 @@ export class NunjucksWorkflowRunner implements WorkflowRunner {
         user: task.spec.user,
       };
 
+      const [decision]: PolicyDecision[] =
+        this.options.permissionApi && task.spec.steps.length
+          ? await this.options.permissionApi.authorizeConditional(
+              [{ permission: actionExecutePermission }],
+              { token: task.secrets?.backstageToken },
+            )
```

Comment:
It's been so long since I last worked with the framework, I've forgotten how this works 😱 just for my own understanding, can you explain why we need to reach out to the permission backend from the scaffolder backend here?

Reply:
before an action is executed, we need to check whether the action is authorized. Here we just get back a decision and then in the `executeStep` method we evaluate the decision with the data coming from each specific action.
We invoke the  `authorizeConditional` here in order to avoid multiple calls to the permission backend, since the data returned as decision is expected to be the same.